README.adoc

```diff
@@ -2,25 +2,151 @@
-them to Cloud Foundry, to Kubernetes, and on your local machine.
+=== Stream Processing Application Overview
+
+In this guide, we will demonstrate the capability of Spring Cloud Stream.  We will create 3 applications to showcase the different
+functionalities of Spring Cloud Stream.
+
+- name-source : publish a String at a regular interval to initiate the stream.  For this example, publish a name as a String.
+- name-processor : consume the String published by name-source and transform the data in some way.  Publish the results to a different exchange.
+For this example, use the name String to create a record and add a timestamp.
+- name-sink : consume the results from name-processor and perform an action.  In this case, print out the results to standard out.
+
+For this example the names of the applications follow the Spring Cloud Stream concepts (`Source`, `Processor`, `Sink`).  These concepts map
+to the logical equivalent of the Java 8 functions (`Supplier`, Function, Consumer respectively).
+While Spring Cloud Stream can support a Function(s) in a Source  as well as Sink via function composition, we will have 3 separate applications to demonstrate how each can work as an independent application.
+
+image::stream-applications-layered-architecture.png[]
+
+In this guide, we will work from the back to the front.  That is, we will first build the Sink application, followed by the Processor, and
+finally build the Source.  We will use the RabbitMQ dashboard UI to test each component as we go.
+
+=== Install Prerequisite - RabbitMQ
+
+In order to use Spring Cloud Stream functionality, we need to ensure that a message broker is accessible.  For this guide we will use RabbitMQ.
+If a local Docker environment is found, the following command can start RabbitMQ
+```shell
+docker run -d --hostname my-rabbit --name some-rabbit -p 15672:15672 -p 5672:5672 rabbitmq:3-management
+```
+
+The result is that RabbitMQ should be link:http://localhost:15672/[accessible locally] with the username/password of guest/guest.
+
+
+== Sink Application
+
+The sink (`java.util.function.Consumer`) is defined in `NameSinkConfiguration` as
+
+====
+[source,java,tabsize=2,indent=0]
+----
+include::name-sink/src/main/java/com/example/stream/namesink/NameSinkConfiguration.java[tag=namesink]
+----
+====
+
+Starting this application without configuring the exchange name will automatically generate an exchange in RabbitMQ titled `nameSink-in-0`.
+We want to customize this so that we can later hook our processor to our sink.
+
+====
+[source,tabsize=2,indent=0]
+----
+include::name-sink/src/main/resources/application.properties[tag=sinkexchangeconfig]
+----
+====
+
+
+To test the sink we will manually publish a JSON message representing a `Person` record to the newly generated exchange:
+
+image::rabbitmq-sink.png[]
+
+And in our application logs we should see the data outputted:
```

Comment:
Change "And" to "Also," (including the comma).

(It's not strictly ungrammatical to start a sentence with a preposition - people do it all the time. However, technical documentation is formal, so we avoid that.)